test/gurka/test_route.cc

```diff
@@ -928,4 +928,8 @@ TEST(AlgorithmTestTrivial, unidirectional_regression) {
                                 {"/date_time/value", "2111-11-11T11:11"},
                             });
   gurka::assert::raw::expect_path(result, {"AB"});
+
+  // again with reverse a* search direction
+  result = gurka::do_action(valhalla::Options::route, map, {"A", "3"}, "bikeshare");
+  gurka::assert::raw::expect_path(result, {"AB"});
```

Comment:
this test doesnt fail anyway even without the changes  to the node snap edge removal. we need to figure out why because it might tell us something else is wrong with unidirectional astar